modules/rostests/apitests/shell32/ShellState.cpp

```diff
@@ -0,0 +1,226 @@
+/*
+ * PROJECT:         ReactOS API tests
+ * LICENSE:         LGPLv2.1+ - See COPYING.LIB in the top level directory
+ * PURPOSE:         Test for SHELLSTATE
+ * PROGRAMMERS:     Katayama Hirofumi MZ <katayama.hirofumi.mz@gmail.com>
+ */
+#include "shelltest.h"
+
+#define NDEBUG
+#include <debug.h>
+#include <shellutils.h>
+#include <stdio.h>
+
+/* [HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Explorer] */
+/* The contents of RegValue ShellState. */
+typedef struct REGSHELLSTATE
+{
+    DWORD dwSize;
+    SHELLSTATE ss;
+} REGSHELLSTATE, *PREGSHELLSTATE;
+
+static void dump(const char *name, const void *ptr, size_t siz)
+{
+    const BYTE *pb;
+    printf("%s: ", name);
+
+    pb = (const BYTE *)ptr;
+    while (siz--)
+    {
+        printf("%02X ", *pb++);
+    }
+    printf("\n");
+}
+
+static int read_key(REGSHELLSTATE *prss)
+{
+    HKEY hKey;
+    LONG result;
+    DWORD cb;
+    static const LPCWSTR s_pszExplorer =
+        L"Software\\Microsoft\\Windows\\CurrentVersion\\Explorer";
+
+    memset(prss, 0, sizeof(*prss));
+
+    result = RegOpenKeyExW(HKEY_CURRENT_USER, s_pszExplorer, 0, KEY_READ, &hKey);
+    ok(result == ERROR_SUCCESS, "\n");
+    ok(hKey != NULL, "\n");
+
+    if (result != ERROR_SUCCESS || !hKey)
+    {
+        skip("RegOpenKeyEx failed: %ld\n", result);
+        return 1;
+    }
+
+    cb = sizeof(*prss);
+    result = RegQueryValueExW(hKey, L"ShellState", NULL, NULL, LPBYTE(prss), &cb);
+    RegCloseKey(hKey);
+
+    ok(result == ERROR_SUCCESS, "\n");
+    if (result != ERROR_SUCCESS)
+    {
+        skip("RegQueryValueEx failed: %ld\n", result);
+        return 2;
+    }
+
+    return 0;
+}
+
+static int dump_pss(SHELLSTATE *pss)
+{
+    dump("SHELLSTATE", pss, sizeof(*pss));
+    printf("\n");
+    return 0;
+}
+
+START_TEST(ShellState)
+{
+    OSVERSIONINFO osinfo;
+    REGSHELLSTATE rss;
+    SHELLSTATE *pss;
+    SHELLFLAGSTATE FlagState;
+    LPBYTE pb;
+    int ret;
+
+    printf("GetVersion(): 0x%08lX", GetVersion());
+    printf("\n");
+
+    osinfo.dwOSVersionInfoSize = sizeof(osinfo);
+    GetVersionEx(&osinfo);
+    printf("osinfo.dwMajorVersion: 0x%08lX\n", osinfo.dwMajorVersion);
+    printf("osinfo.dwMinorVersion: 0x%08lX\n", osinfo.dwMinorVersion);
+    printf("osinfo.dwBuildNumber: 0x%08lX\n", osinfo.dwBuildNumber);
+    printf("osinfo.dwPlatformId: 0x%08lX\n", osinfo.dwPlatformId);
+    printf("\n");
+
+    printf("WINVER: 0x%04X\n", WINVER);
+    printf("_WIN32_WINNT: 0x%04X\n", _WIN32_WINNT);
+    printf("_WIN32_IE: 0x%04X\n", _WIN32_IE);
+    printf("NTDDI_VERSION: 0x%08X\n", NTDDI_VERSION);
+
+#ifdef UNICODE
```

Comment:
This does not seem useful. It will not change, and your test should not depend on it.

Reply:
Deleted.